service/history/timerQueueActiveProcessor.go

```
@@ -547,7 +547,9 @@ func (t *timerQueueActiveProcessorImpl) processDecisionTimeout(
 		metricScopeWithDomainTag.IncCounter(metrics.StartToCloseTimeoutCounter)
 		if decision.Attempt == task.ScheduleAttempt {
 			// Add a decision task timeout event.
-			msBuilder.AddDecisionTaskTimedOutEvent(scheduleID, decision.StartedID)
+			if _, err := msBuilder.AddDecisionTaskTimedOutEvent(scheduleID, decision.StartedID); err != nil {
+				t.logger.Error("failed to add task time out event", tag.Error(err))
```

@carlcortright please return the err